edx/edx-enterprise
# ENT-641 Add functionality for transmitting courses for a specific enterprise

**Description:** Add functionality for transmitting courses for a specific enterprise.

_**Note:** The logs during transmission of courses metadata can be viewed on splunk for stage and prod._

**JIRA:** [ENT-641](https://openedx.atlassian.net/browse/ENT-641)

**Dependencies:** N/A

**Merge deadline:** N/A

**Installation instructions:** Install `edx-enterprise` from this branch `zub/ENT-641-publish-courses-metadata-for-specific-enterprises` in `edx-platform`.

**Testing instructions:**

1. Add an enterprise with catalogs
2. Add configurations for integrated channels for Degreed and SAP Success Factors
3. Open django admin for Enterprise Customers.
4. Open detail page of above enterprise and verify that you can see a new action button `Transmit Courses Metadata` in top "Change Enterprise Customer" bar at right side along with existing action button `Manage Learners`.
5. Click on the action button `Transmit Courses Metadata`, this will take you to "Transmit Courses Metadata" page.
6. Provide username of the enterprise channel worker user which we use for transmitting courses metadata for the integrated channels and then verify that clicking on the `Submit` button triggers the django management command `transmit_course_metadata` for the current enterprise and with the provided catalog user.
7. Also verify that providing empty value or invalid username gives validation error.

<img width="1049" alt="screen shot 2018-02-08 at 6 52 17 pm" src="https://user-images.githubusercontent.com/5072991/35976300-83a9515c-0d01-11e8-88e2-10fbdb564aa6.png">
<img width="821" alt="screen shot 2018-02-09 at 5 23 27 pm" src="https://user-images.githubusercontent.com/5072991/36027677-166a1b4a-0dbe-11e8-9592-fb881f1ef2ff.png">
<img width="820" alt="screen shot 2018-02-09 at 5 23 42 pm" src="https://user-images.githubusercontent.com/5072991/36027688-2001f98e-0dbe-11e8-8bd2-3f56949affa9.png">



**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
